File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        n = len(nums)
        curr = sum(nums[:k])
        print(curr)
        ret = curr/k
        l = 0
        for r in range(k,n):
            curr+=nums[r]
            curr-=nums[l]
            l+=1
            ret = max(curr/k,ret)
        return ret

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.ts
function findMaxAverage(nums: number[], k: number): number {
    let curr = 0;
    for (let i =0;i<k;i++){
        curr +=nums[i]
    }
    let ret = curr/k;
    let l = 0
    for(let r = k;r<nums.length;r++){
        curr+= nums[r]
        curr-= nums[l]
        l++
        ret = Math.max(curr/k,ret) 
    }
    return ret
};